### PR TITLE
Update changed_organization.html

### LINKED ISSUE #5771: Organization changes do not have detailed Activity Stream UI
**CKAN version**
2.9.1

**Describe the bug**
One of the major features in 2.9 is the detailed Dataset Activity Stream.  Even though Organization changes are also stored in the `activity` table, it only logs that the Organization has been updated.

**Steps to reproduce**
Make changes to an Organization and check the Activity Stream.

**Expected behavior**
The Activity Stream should at least show the diff, similar to how its done with Datasets, as the Organization metadata is stored in `activity.data` column.

### PR DESCRIPTION
**Pull request by Gauravp-NEC**
Fix for Issue 5771

Fixes #5771 

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by Gauravp-NEC** [comment]
@Zharktas , Please review this PR. Thankyou :)

**Comment by Gauravp-NEC** [comment]
@Zharktas , Please review this PR as it is open for a long time.

**Comment by Gauravp-NEC** [comment]
@Zharktas , Please review this PR as it is open for a long time.

File: ckan/templates/snippets/organization_changes/description.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+              )|safe,
+              old_description = change.old_description,
+              new_description = change.new_description
+        ) }}
+
+    {% elif change.method == "add" %}
+
+      {{ _('Updated description of {pkg_link} to <br class="line-height2"><blockquote>{new_description}</blockquote>').format(
+              pkg_link = '<a href="{pkg_url}">{pkg_name}</a>'.format(
+                pkg_url = h.url_for(controller='organization',
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/description.html:19]: Should be url_for to named route instead on controller/action as controllers are soon removed. See https://github.com/ckan/ckan/blob/abfda8b401ec370c9c74fc67678dcb38f8c72a26/ckan/templates/snippets/changes/notes.html#L14

File: ckan/templates/snippets/organization_changes/description.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                pkg_name = change.title
+              )|safe,
+              new_description = change.new_description
+          ) }}
+
+    {% elif change.method == "remove" %}
+
+      {{ _('Removed description from {pkg_link}').format(
+            pkg_link = '<a href="{pkg_url}">{pkg_name}</a>'.format(
+              pkg_url = h.url_for(controller='organization',
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/description.html:30]: Should use named route also here.

File: ckan/templates/snippets/organization_changes/image_url.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
+<li>
+  <p>
+    {% if change.method == "change" %}
+
+      {{ _('Changed the image URL of {pkg_link} from {old_link} to {new_link}').format(
+            pkg_link = '<a href="{pkg_image_url}">{pkg_name}</a>'.format(
+              pkg_image_url = h.url_for(controller='organization',
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/image_url.html:7]: Needs named_route.

File: ckan/templates/snippets/organization_changes/image_url.html [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+              new_image_url = "http://" + change.new_image_url if not change.new_image_url[0:4] == "http" else change.new_image_url,
+              new_image_url_name = change.new_image_url
+            )|safe
+      ) }}
+
+    {% elif change.method == "remove" %}
+
+      {{ _('Removed the image URL from {pkg_link}').format(
+            pkg_link = '<a href="{pkg_image_url}">{pkg_name}</a>'.format(
+              pkg_image_url = h.url_for(controller='organization',
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/image_url.html:25]: Needs named_route

File: ckan/templates/snippets/organization_changes/image_url.html [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                                  action='read', id=change.pkg_id),
+              pkg_name = change.title
+            )|safe
+      ) }}
+
+    {% elif change.method == "add" %}
+
+      {{ _('Changed the image URL of {pkg_link} to {new_link}').format(
+            pkg_link = '<a href="{pkg_image_url}">{pkg_name}</a>'.format(
+              pkg_image_url = h.url_for(controller='organization',
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/image_url.html:35]: Needs named_route

File: ckan/templates/snippets/organization_changes/title.html [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<li>
+  <p>
+    {{ _('Changed title to {title_link} (previously {old_title})').format(
+          title_link = '<a href="{new_url}">{new_title}</a>'.format(
+            new_url = h.url_for(controller='organization', action='read', id=change.id),
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/title.html:5]: Needs named_route

**Review by Zharktas — CHANGES_REQUESTED** [review]
@Gauravp-NEC seems mostly fine. url_for's should use named_routes instead of controller/action parameters as controllers are soon removed. Custom fields are shown on the changes view, these should be added, they are named extras in the code. The name of the template organization/organization_changes.html could be just organization/changes.html.

**Comment by Gauravp-NEC** [comment]
@Zharktas, I have done rework in commit  https://github.com/ckan/ckan/pull/5985/commits/ba23409ba8184ca112e0a60bf98cdf8e6dc3530d. If it looks good please merge this PR. Thankyou

File: ckan/templates/snippets/organization_changes/description.html [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<li>
+  <p>
+    {% if change.method == "change" %}
+
+      {{ _('Updated description of {pkg_link} from <br class="line-height2"><blockquote>{old_description}</blockquote> to <br class="line-height2"><blockquote>{new_description}</blockquote>').format(
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/description.html:5]: pkg_link could be renamed to org_link for clarity.

File: ckan/templates/snippets/organization_changes/description.html [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        pkg_link = h.nav_link(change.title, named_route='organization.read', id=change.pkg_id),
+        old_description = change.old_description,
+        new_description = change.new_description
+        ) }}
+
+    {% elif change.method == "add" %}
+
+      {{ _('Updated description of {pkg_link} to <br class="line-height2"><blockquote>{new_description}</blockquote>').format(
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/description.html:13]: pkg_link could be renamed to org_link for clarity.

File: ckan/templates/snippets/organization_changes/description.html [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        pkg_link = h.nav_link(change.title, named_route='organization.read', id=change.pkg_id),
+        new_description = change.new_description
+        ) }}
+
+    {% elif change.method == "remove" %}
+
+      {{ _('Removed description from {pkg_link}').format(
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/description.html:20]: pkg_link could be renamed to org_link for clarity.

File: ckan/templates/snippets/organization_changes/image_url.html [hunk 10]
```diff
@@ -0,0 +1,30 @@
+<li>
+  <p>
+    {% if change.method == "change" %}
+
+      {{ _('Changed the image URL of {pkg_link} from {old_link} to {new_link}').format(
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/image_url.html:5]: pkg_link could be renamed to org_link for clarity.

File: ckan/templates/snippets/organization_changes/image_url.html [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        pkg_link = h.nav_link(change.title, named_route='organization.read', id=change.pkg_id),
+        old_link = h.link_to(change.old_image_url, "http://" + change.old_image_url if not change.old_image_url[0:4] == "http" else change.old_image_url),
+        new_link = h.link_to(change.new_image_url, "http://" + change.new_image_url if not change.new_image_url[0:4] == "http" else change.new_image_url),
+        ) }}
+
+    {% elif change.method == "remove" %}
+
+      {{ _('Removed the image URL from {pkg_link}').format(
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/image_url.html:13]: pkg_link could be renamed to org_link for clarity.

File: ckan/templates/snippets/organization_changes/image_url.html [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+        pkg_link = h.nav_link(change.title, named_route='organization.read', id=change.pkg_id),
+        ) }}
+
+    {% elif change.method == "add" %}
+
+      {{ _('Changed the image URL of {pkg_link} to {new_link}').format(
```
Comment (Zharktas) [inline ckan/templates/snippets/organization_changes/image_url.html:19]: pkg_link could be renamed to org_link for clarity.

**Review by Zharktas — CHANGES_REQUESTED** [review]
Variable names are a minor thing.

But if you could add those custom field via extras to changes view, this would be good to merge. 

Now if you only update custom fields, you get changes link in activity list but the view doesn't show that anything changed.

**Comment by Gauravp-NEC** [comment]
@Zharktas , I have changed the "pkg_link" to "org_link" in commit https://github.com/ckan/ckan/pull/5985/commits/c98370b4991e5ac798879420fc77dab366d9f864.

File: ckan/views/group.py [hunk 13]
```diff
@@ -507,6 +508,132 @@ def activity(id, group_type, is_organization, offset=0):
...
+        log.info(u'Activity not found: {} - {}'.format(str(e), activity_id))
+        return base.abort(404, _(u'Activity not found'))
+    except NotAuthorized:
+        return base.abort(403, _(u'Unauthorized to view activity data'))
+
+    # 'group_dict' needs to go to the templates for page title & breadcrumbs.
+    # Use the current version of the package, in case the name/title have
+    # changed, and we need a link to it which works
+    group_id = activity_diff[u'activities'][1][u'data'][u'group'][u'id']
+    current_group_dict = get_action(u'organization_show')(
```
Comment (Zharktas) [inline ckan/views/group.py:535]: This should use group_type as part  of the action call.
```suggestion
    current_group_dict = get_action(group_type + u'_show')(
```

File: ckan/views/group.py [hunk 14]
```diff
@@ -507,6 +508,132 @@ def activity(id, group_type, is_organization, offset=0):
...
+        context, {u'id': group_id})
+    group_activity_list = get_action(u'organization_activity_list')(
```
Comment (Zharktas) [inline ckan/views/group.py:537]:
```suggestion
    group_activity_list = get_action(group_type + u'_activity_list')(
```

File: ckan/views/group.py [hunk 15]
```diff
@@ -507,6 +508,132 @@ def activity(id, group_type, is_organization, offset=0):
...
+        }
+    )
+
+
+def changes_multiple(is_organization, group_type=None):
+    '''
+    Called when a user specifies a range of versions they want to look at
+    changes between. Verifies that the range is valid and finds the set of
+    activity diffs for the changes in the given version range, then
+    re-renders organization_changes.html with the list.
```
Comment (Zharktas) [inline ckan/views/group.py:559]:
```suggestion
    re-renders changes.html with the list.
```

File: ckan/views/group.py [hunk 16]
```diff
@@ -507,6 +508,132 @@ def activity(id, group_type, is_organization, offset=0):
...
+
+        diff_list.append(activity_diff)
+
+        if activity_diff['activities'][0]['id'] == old_id:
+            done = True
+        else:
+            current_id = activity_diff['activities'][0]['id']
+
+    group_id = diff_list[0][u'activities'][1][u'data'][u'group'][u'id']
+    current_group_dict = get_action(u'organization_show')(
```
Comment (Zharktas) [inline ckan/views/group.py:621]:
```suggestion
    current_group_dict = get_action(group_type + u'_show')(
```

File: ckan/views/group.py [hunk 17]
```diff
@@ -507,6 +508,132 @@ def activity(id, group_type, is_organization, offset=0):
...
+        context, {u'id': group_id})
+    group_activity_list = get_action(u'organization_activity_list')(context, {
```
Comment (Zharktas) [inline ckan/views/group.py:623]:
```suggestion
    group_activity_list = get_action(group_type + u'_activity_list')(context, {
```

**Review by Zharktas — COMMENTED** [review]
group_type should be used so that the code doesn't need to be duplicated for groups.

**Comment by Gauravp-NEC** [comment]
@Zharktas , Please merge this PR if it looks good. Thankyou

**Review by Zharktas — APPROVED** [review]

**Pull request merged**